File: advertising_sdk/src/main/java/com/adadapted/android/sdk/ext/json/JsonSessionRequestBuilder.java
package com.adadapted.android.sdk.ext.json;

import android.util.Log;

import com.adadapted.android.sdk.core.device.DeviceInfo;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.Date;

public class JsonSessionRequestBuilder {
    private static final String LOGTAG = JsonSessionRequestBuilder.class.getName();

    private static final String APP_ID = "app_id";
    private static final String UDID = "udid";
    private static final String BUNDLE_ID = "bundle_id";
    private static final String BUNDLE_VERSION = "bundle_version";
    private static final String DEVICE = "device";
    private static final String DEVICE_UDID = "device_udid";
    private static final String OS = "os";
    private static final String OSV = "osv";
    private static final String LOCALE = "locale";
    private static final String TIMEZONE = "timezone";
    private static final String DATETIME = "datetime";
    private static final String CARRIER = "carrier";
    private static final String DH = "dh";
    private static final String DW = "dw";
    private static final String DENSITY = "density";
    private static final String ALLOW_RETARGETING = "allow_retargeting";
    private static final String SDK_VERSION = "sdk_version";
    private static final String PARAMS = "init_params";

    public JSONObject buildSessionInitRequest(final DeviceInfo deviceInfo) {
        final JSONObject json = new JSONObject();

        try {
            json.put(APP_ID, deviceInfo.getAppId());
            json.put(UDID, deviceInfo.getUdid());
            json.put(BUNDLE_ID, deviceInfo.getBundleId());
            json.put(BUNDLE_VERSION, deviceInfo.getBundleVersion());
            json.put(DEVICE, deviceInfo.getDevice());
            json.put(DEVICE_UDID, deviceInfo.getDeviceUdid());
            json.put(OS, deviceInfo.getOs());
            json.put(OSV, deviceInfo.getOsv());
            json.put(LOCALE, deviceInfo.getLocale());
            json.put(TIMEZONE, deviceInfo.getTimezone());
            json.put(CARRIER, deviceInfo.getCarrier());
            json.put(DH, deviceInfo.getDh());
            json.put(DW, deviceInfo.getDw());
            json.put(DENSITY, deviceInfo.getDensity());
            json.put(DATETIME, new Date().getTime());
            json.put(ALLOW_RETARGETING, deviceInfo.isAllowRetargetingEnabled() ? 1 : 0);
            json.put(SDK_VERSION, deviceInfo.getSdkVersion());
            json.put(PARAMS, new JSONObject(deviceInfo.getParams()));
        }
        catch(JSONException ex) {
            Log.d(LOGTAG, "Problem converting to JSON.", ex);
        }

        return json;
    }
}

File: advertising_sdk/src/main/java/com/adadapted/android/sdk/ext/json/JsonSessionRequestBuilder.kt
package com.adadapted.android.sdk.ext.json

import android.util.Log
import com.adadapted.android.sdk.core.device.DeviceInfo
import org.json.JSONException
import org.json.JSONObject
import java.util.Date

class JsonSessionRequestBuilder {
    fun buildSessionInitRequest(deviceInfo: DeviceInfo): JSONObject {
        val json = JSONObject()
        try {
            json.put(APP_ID, deviceInfo.appId)
            json.put(UDID, deviceInfo.udid)
            json.put(BUNDLE_ID, deviceInfo.bundleId)
            json.put(BUNDLE_VERSION, deviceInfo.bundleVersion)
            json.put(DEVICE_NAME, deviceInfo.device)
            json.put(DEVICE_UDID, deviceInfo.deviceUdid)
            json.put(DEVICE_OS, deviceInfo.os)
            json.put(DEVICE_OSV, deviceInfo.osv)
            json.put(DEVICE_LOCALE, deviceInfo.locale)
            json.put(DEVICE_TIMEZONE, deviceInfo.timezone)
            json.put(DEVICE_CARRIER, deviceInfo.carrier)
            json.put(DEVICE_HEIGHT, deviceInfo.dh)
            json.put(DEVICE_WIDTH, deviceInfo.dw)
            json.put(DEVICE_DENSITY, Integer.valueOf(deviceInfo.density).toString())
            json.put(ALLOW_RETARGETING, deviceInfo.isAllowRetargetingEnabled)
            json.put(CREATED_AT, Date().time)
            json.put(SDK_VERSION, deviceInfo.sdkVersion)
            json.put(PARAMS, JSONObject(deviceInfo.params))
        } catch (ex: JSONException) {
            Log.d(LOGTAG, "Problem converting to JSON.", ex)
        }
        return json
    }

    companion object {
        private val LOGTAG = JsonSessionRequestBuilder::class.java.name
        private const val APP_ID = "app_id"
        private const val UDID = "udid"
        private const val BUNDLE_ID = "bundle_id"
        private const val BUNDLE_VERSION = "bundle_version"
        private const val DEVICE_NAME = "device_name"
        private const val DEVICE_UDID = "device_udid"
        private const val DEVICE_OS = "device_os"
        private const val DEVICE_OSV = "device_osv"
        private const val DEVICE_LOCALE = "device_locale"
        private const val DEVICE_TIMEZONE = "device_timezone"
        private const val DEVICE_CARRIER = "device_carrier"
        private const val DEVICE_HEIGHT = "device_height"
        private const val DEVICE_WIDTH = "device_width"
        private const val DEVICE_DENSITY = "device_density"
        private const val ALLOW_RETARGETING = "allow_retargeting"
        private const val CREATED_AT = "created_at"
        private const val SDK_VERSION = "sdk_version"
        private const val PARAMS = "params"
    }
}